test/parallel/test-gc-net-timeout.js

```diff
@@ -3,12 +3,14 @@
 // just like test-gc-http-client-timeout.js,
 // but using a net server/client instead
 
-require('../common');
+const common = require('../common');
 const onGC = require('../common/ongc');
 const assert = require('assert');
 const net = require('net');
 const os = require('os');
 
+common.setHappyEyeballsEnabled(false);
```

Comment:
Yes, see https://github.com/nodejs/node/pull/44731#issuecomment-1260760298

Reply:
If switching it on by default breaks our tests, then it will likely break external packages or at least their tests as well.

Reply:
Does this impl actually establish more than one TCP connection? Isn't the idea to only send SYN to all addresses and to only send the ACK to the first address that replies with SYN-ACK?

Reply:
About failing test yes, it might break them. We will need to communicate properly in the release nodes, but there's nothing we can do about it.

About SYN/ACK, I this this is handle by OS (via `uv_tcp_connect`) so we don't have control over it. Am I right?

Reply:
> About failing test yes, it might break them. We will need to communicate properly in the release nodes, but there's nothing we can do about it.

If this breaks a popular package that is not actively maintained, it could still be problematic. I remember how https://github.com/nodejs/node/pull/41431 broke https://github.com/indutny/node-ip. Looking at the downstream issue - https://github.com/indutny/node-ip/issues/113 it can be seen that it took a considerable amount of time for the maintainer to respond.

Reply:
I can relate.
But I think fixing the connection problem in this case has higher priority. Nonetheless, this PR has the option disable so nothing will break out of the blue.
We will have to review it when we flip the default.

Reply:
:+1:

> We will have to review it when (if) we flip the default.

Yea, just mentioned my concern here because [the question for the TSC was regarding flipping the default](https://github.com/nodejs/TSC/issues/1287#issuecomment-1259178815) and this PR had the `tsc-agenda` label :)

Reply:
Coo! Solving this!